cmd/compose/ps.go

```diff
@@ -113,7 +113,7 @@ func runPs(ctx context.Context, dockerCli command.Cli, backend api.Service, serv
 
 	containers, err := backend.Ps(ctx, name, api.PsOptions{
 		Project:  project,
-		All:      opts.All,
+		All:      opts.All || containsStatus(opts.Status, "exited"),
```

Comment:
AFAICT as long as we filter by status we can force `All`, as irrelevant containers will anyway be filtered

Reply:
You are right but incase where users explicitly set it to false we will not be able to see the exited containers.
IMO either we should throw an error in these cases or have the default behaviour as in the change.
Open to your opinion on this.

Reply:
`opts.All || opts.Status != ""` sounds enough to me. We could check `--all` flag was not intentionally set to `false` in command `PreRun` to prevent misuse, but as a corner case with no benefit I wouldn't worry about it